config/routes.rb

```diff
@@ -78,6 +78,15 @@
       end
     end
 
+    resources :messages, :path => "user/messages", :constraints => { :id => /\d+/ }, :only => [:create, :show, :destroy], :controller => "messages", :as => :api_messages do
+      collection do
+        get "inbox"
+        get "outbox"
+      end
+    end
+
+    post "/user/messages/:id" => "messages#update", :as => :api_message_update
```

Comment:
Why can't this be done by allowing `:update` on the resource block above?

Reply:
Adding `:update` to resources above would add PUT and PATCH routes. PUT implies that the whole message would be modified, and I didn't see any PATCH ops in [API_v0.6 spec](https://wiki.openstreetmap.org/wiki/API_v0.6)

The API proposal suggested POST with single parameter: `read_status` with possible values of `true` and `false`, similar to how  "POST /api/0.6/changeset/comment/#comment_id/hide" works.

Reply:
Technically I guess `PATCH` would be the correct thing but as you say it's probably not consistent with the API to date and I haven't managed to find a way to move that route inside the resource block while keeping it as POST unless @gravitystorm has some ideas?